solidity/contracts/ERC20WithData.sol

```diff
@@ -9,26 +9,25 @@ import '@openzeppelin/contracts/utils/Context.sol';
     @dev Mintable+burnable form of ERC20 with data event support.
 */
 contract ERC20WithData is Context, ERC20 {
-    constructor(string memory name, string memory symbol) public ERC20(name, symbol) {}
+    constructor(string memory name, string memory symbol) ERC20(name, symbol) {}
 
     function mintWithData(
         address to,
         uint256 amount,
         bytes calldata data
-    ) external virtual {
+    ) public {
         _mint(to, amount);
     }
 
     function transferWithData(
-        address from,
         address to,
         uint256 amount,
         bytes calldata data
-    ) external virtual {
-        _transfer(from, to, amount);
```

Comment:
Why did you change the signature here? I think we should keep the same parameters as the base method (`from, to, amount`) and pass them through.

Reply:
Oh I see, you were calling a private method without approval checks.

Perhaps we should be proxying to `transferFrom` instead?